### PR TITLE
[instrument_builder] Let mindate and maxdate options be removable

### PR DESCRIPTION
**Pull request by zaliqarosli**
## Brief summary of changes

This PR allows the Start year and End year values for Date elements be emptied

#### Testing instructions (if applicable)

1. On branch 23.0-release, add a Date element and notice that once you enter a value for both Start year and End year, you are not able to make the fields empty
2. Save the instrument with one Date element without min/max values, and another Date element with min/max values.
3. On this PR's branch, add a Date element and notice that you can make it empty after inserting a value. Make sure you run `make dev` after changing branches.
4. Save the instrument with the same elements as step 2.
5. Compare the two saved linst files and see that they are identical. This PR should only affect the front-end instrument builder. Please report if otherwise.

#### Link(s) to related issue(s)

* Resolves https://github.com/aces/Loris/issues/6567

### CODE REVIEW

File: modules/instrument_builder/jsx/react.questions.js [hunk 1]
```diff
@@ -336,8 +336,8 @@ class DateOptions extends Component {
   // Render the HTML
   render() {
     // Truncate off the month and day from the date to only have the year.
-    let minYear = this.props.element.Options.MinDate.split('-')[0];
```
Comment (driusan) [inline modules/instrument_builder/jsx/react.questions.js:339]: both the comment above and the variable names are wrong.. why is this change being done? It seems unrelated to the title and I'm not sure of the impact of it on existing instruments.

Comment (zaliqarosli) [reply modules/instrument_builder/jsx/react.questions.js:339]: there's no impact on existing instruments i don't think. only within the instrument builder, the minyear and maxyear are being joined by 01-01 and 31-12 to make a full date which is unnecessary because linst only takes the year anyways. this line above that i'm removing then removes the month and day to only take the year. so i'm removing any addition/subtraction of day and month

Comment (driusan) [reply modules/instrument_builder/jsx/react.questions.js:339]: what about when loading existing instruments?

Comment (zaliqarosli) [reply modules/instrument_builder/jsx/react.questions.js:339]: hmmm something seems to be going wrong when you reload the linst and resave, i'll fix that now

Comment (zaliqarosli) [reply modules/instrument_builder/jsx/react.questions.js:339]: @driusan that is a separate bug reported here https://github.com/aces/Loris/issues/6669. this PR makes no changes to the linst file output

Comment (zaliqarosli) [reply modules/instrument_builder/jsx/react.questions.js:339]: > what about when loading existing instruments?

existing instruments only carry the year, so i don't understand why having the day and month within the instrument builder affects that

**Comment by driusan** [comment]
@zaliqarosli Travis doesn't seem to be started on this, maybe it needs to be rebased?

**Comment by driusan** [comment]
This passed Travis if you click the link, the icon just didn't update.
